cmd/tailscale/cli/up.go

```diff
@@ -436,10 +436,10 @@ func runUp(ctx context.Context, cmd string, args []string, upArgs upArgsT) (retE
 	}
 
 	if len(prefs.AdvertiseRoutes) > 0 {
-		if err := localClient.CheckIPForwarding(context.Background()); err != nil {
+		if err := localClient.CheckIPForwarding(ctx); err != nil {
 			warnf("%v", err)
 		}
-		if err = localClient.CheckUDPGROForwarding(context.Background()); err != nil {
+		if err = localClient.CheckUDPGROForwarding(ctx); err != nil {
```

Comment:
:= here like L439

Do we only want to do this if runtime.GOOS == "linux"? Doing another HTTP request to localhost (two back to back!) seems a little silly. Maybe a TODO to combine these into one CheckStuff call?

But I guess this is only for AdvertisedRoutes > 0 so it won't happen for most users at least.

Or maybe the LocalClient method itself should return early with nil if GOOS != "linux" and keep this code without the conditional.

Reply:
> Do we only want to do this if runtime.GOOS == "linux"?

For now, yes. Probably pretty unlikely to do this for non-Linux in the future.

> Doing another HTTP request to localhost (two back to back!) seems a little silly. Maybe a TODO to combine these into one CheckStuff call?

Fair, compressing them at just the HTTP layer is reasonable. Added a TODO for myself.

> Or maybe the LocalClient method itself should return early with nil if GOOS != "linux" and keep this code without the conditional.

I gated it here for now. Once they are compressed I'll remove the condition.

Updates in https://github.com/tailscale/tailscale/pull/10071/commits/abd9ef8d7f5303125cea337e17b7bf3dfc5a5772